ethers-contract/src/call.rs

```diff
@@ -211,3 +218,18 @@ where
             .map_err(ContractError::MiddlewareError)
     }
 }
+
+/// [`ContractCall`] can be turned into [`Future`] automatically with `.await`.
+/// Defaults to calling [`ContractCall::call`].
+impl<M, D> IntoFuture for ContractCall<M, D>
+where
+    M: Middleware + 'static,
+    D: Detokenize + 'static,
```

Comment:
For some reason, compiler refuses to accept code without `'static` lifetimes.

Reply:
The reason is that the future created is `'static` so it needs to know that neither the `Middleware` nor the `Detokenize` will have a shorter lifetime than `'static`. The best way to express this is probably as follows:

```
impl<M, D> IntoFuture for ContractCall<M, D>
where
    Self: 'static,
    M: Middleware,
    D: Detokenize,
```

Reply:
Moved to `Self`.